config/src/test/scala/com/typesafe/config/impl/ConfigSubstitutionTest.scala

```diff
@@ -924,7 +924,7 @@ class ConfigSubstitutionTest extends TestUtils {
 
     @Test
     def substSelfReferenceIndirect() {
-        val obj = parseObject("""a=1, b=${a}, a=${b}""")
+        val obj = parseObject("""b=${a}, a=${b}""")
```

Comment:
It passes on Java 7 and fails on Java 8. Should you be able to specify a key twice like this? It seems pretty weird. Maybe we should change this test as I have here, but also add a new test that disallows having a key twice and throws an exception?